### PR TITLE
More metrics, remove hard-coded prometheus

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: server/log_operation_manager.go [hunk 1]
```diff
@@ -19,20 +19,39 @@ import (
...
+	knownLogs    monitoring.Gauge
+	resignations monitoring.Counter
+	isMaster     monitoring.Gauge
+)
+
+func createMetrics(mf monitoring.MetricFactory) {
+	if mf == nil {
+		mf = monitoring.InertMetricFactory{}
+	}
+	knownLogs = mf.NewGauge("known_logs", "Set to 1 for known logs", logIDLabel)
```
Comment (Martin2112) [inline server/log_operation_manager.go]: Can you just clarify what known_logs is? Looks like the logs this signer is master for?

File: server/log_operation_manager.go [hunk 2]
```diff
@@ -112,6 +132,7 @@ func (er *electionRunner) Run(ctx context.Context) {
 		}
 		glog.V(1).Infof("%d: Now, I am the master", er.logID)
```
Comment (Martin2112) [inline server/log_operation_manager.go]: A Dr Who quote here would be good.

Comment (daviddrysdale) [inline server/log_operation_manager.go]: Wrong cultural reference -- see L108...

File: server/log_operation_manager.go (hunk 1 above)
Comment (daviddrysdale) [inline server/log_operation_manager.go]: No, it's logs known about whether or not we're master (I've clarified the help).

My thought is that `known_logs` on the logsigner and the ctfe should be equal (for a CT-only Trillian instance).  
Likewise, if we sum `is_master` across instances it should always be 1 for each logID, and the sum across logIDs should equal the sum of `known_logs`.

Comment (Martin2112) [inline server/log_operation_manager.go]: OK known_logs implied that but wasn't 100% clear. The extra help text is useful.

**Pull request merged**